### PR TITLE
Fix ror & mask_ror x86 backend

### PR DESCRIPTION
**Pull request by SadiinsoSnowfall**
Fixing ror & mask_ror impl in eve
first commit should not pass CI by design

### CODE REVIEW

**Comment by jtlap** [comment]
perhaps you can add doc based on shl file pattern

File: include/eve/module/core/regular/impl/rotl.hpp [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 namespace eve::detail
 {
-  template<typename T, typename S, callable_options O>
+  template<unsigned_value T, typename S, callable_options O>
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/rotl.hpp:18]: this normally useless as the cocnept checkign is done on operator()

File: include/eve/module/core/regular/impl/rotl.hpp [hunk 2]
```diff
@@ -31,7 +31,7 @@ namespace eve::detail
     else                                                            return apply_over(rotl[o], v, s);
   }
 
-  template<typename T, auto S, callable_options O>
+  template<unsigned_value T, auto S, callable_options O>
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/rotl.hpp:34]: this normally useless as the cocnept checkign is done on operator()

File: include/eve/module/core/regular/rotl.hpp [hunk 3]
```diff
@@ -37,12 +37,64 @@ namespace eve
...
+  //!
+  //!     * `x`:  [argument](@ref eve::unsigned_value) to be rotated.
+  //!     * `n`:  [shift](@ref eve::integral_value).
+  //!
+  //!    **Return value**
+  //!
+  //!      The [elementwise](@ref glossary_elementwise) bitwise rotation to the right of the first
+  //!      parameter by the second.
+  //!
+  //!      The call `rotl(x, n)` is equivalent to `std::rotl(x, n)` if `x` is an [simd value](@ref
```
Comment (jfalcou) [inline include/eve/module/core/regular/rotl.hpp:78]: sounds weird

File: include/eve/module/core/regular/rotr.hpp [hunk 4]
```diff
@@ -38,12 +38,64 @@ namespace eve
...
+  //!
+  //!     * `x`:  [argument](@ref eve::unsigned_value) to be rotated.
+  //!     * `n`:  [shift](@ref eve::integral_value).
+  //!
+  //!    **Return value**
+  //!
+  //!      The [elementwise](@ref glossary_elementwise) bitwise rotation to the right of the first
+  //!      parameter by the second.
+  //!
+  //!      The call `rotr(x, n)` is equivalent to `std::rotr(x, n)` if `x` is an [simd value](@ref
```
Comment (jfalcou) [inline include/eve/module/core/regular/rotr.hpp:79]: same. we probably means to say : on scalar value OR to say it is done on each lanes.

**Review by jfalcou — CHANGES_REQUESTED** [review]

**Review by jtlap — APPROVED** [review]

**Review by jfalcou — APPROVED** [review]

**Pull request merged**